crates/transaction-emitter-lib/src/cluster.rs

```diff
@@ -37,7 +39,7 @@ impl Cluster {
         mint_key: Ed25519PrivateKey,
         chain_id: ChainId,
         vasp: bool,
-    ) -> Self {
+    ) -> Result<Self> {
```

Comment:
Let's just make this function async instead of using block_on?

Reply:
Yes, I thought about that but that requires a lot of plumbing - not worth the effort IMO. I can follow up on this if you have a strong opinion about it.